srml/contracts/src/exec.rs

```diff
@@ -283,10 +284,14 @@ where
 		}
 	}
 
-	fn nested(&self, overlay: OverlayAccountDb<'a, T>, dest: T::AccountId) -> Self {
+	fn nested(
+		&self,
+		overlay: OverlayAccountDb<'a, T>,
+		dest: T::AccountId,
+		contract_info: Option<ContractInfo<T>>
```

Comment:
as this argument is only used to get the contract_trie_id and this can be misleading for example for the instantiate call then I would argue in favor of having an `Option<TrieId>` instead, thus this is more explicit, instantiate code will call it with None and we can remove the create_contract_info from inside the overlay account db.

Also as a note:
when I tried to reorganize AccountDb a while ago I very didn't like the way trie_id is handle. I thought that overlay should create a pending TrieId which when committing to direct account db will actually create a real one.
Thus I made OverlayAccountDB a trait with an associated type TrieId, and implemented in such a way that it could merge when committing to underlying overlay. But actually this added a bit complexity in the code I felt that handling trie_id by hand would be ok and better.